packages/in_app_purchase/in_app_purchase_ios/lib/src/in_app_purchase_ios_platform.dart

```diff
@@ -158,12 +158,16 @@ class _TransactionObserver implements SKTransactionObserverWrapper {
   Completer? _restoreCompleter;
   late String _receiptData;
 
+  bool _isRestoringTransactions = false;
+  bool _didReceiveRestoredTransactions = false;
```

Comment:
This would be less error-prone as a single state enum rather than two bools; that way it's impossible to have a bug one one is reset but the other isn't. E.g.:
```
enum _TransactionRestoreState {
  notRunning,
  waitingForTransactions,
  receivedTransaction,
}
```